### ISSUE TITLE
vector serialization is not correct

### ISSUE THREAD

**Issue by vangork** [tafia/quick-xml#784]
```rust
use serde::Serialize;

#[derive(Serialize)]
#[serde(rename = "a")]
struct A {
    name: String,
}

#[derive(Serialize)]
struct B {
    tags: Vec<A>,
    name: String,
}

fn main() {
    let b = B {
        tags: vec![A { name: "test".to_string() }, A { name: "test1".to_string() }],
        name: "test".to_string(),
    };

    println!("{}", quick_xml::se::to_string(&b).unwrap());
}
```

serialization of B should be:
```xml
<B>
    <tags>
        <a>
          <name>test</name>
       </a>
       <a>
           <name>test1</name>
       </a>
    </tags>
    <name>test</name>
</B>
```

instead of:
```xml
<B>
    <tags>
       <name>test</name>
    </tags>
    <tags>
        <name>test1</name>
    </tags>
    <name>test</name>
</B>
```

**Comment by Mingun** [comment]
The serialization worked as expected as described in the [documentation](https://docs.rs/quick-xml/latest/quick_xml/de/index.html#mapping-xml-to-rust-types). Type names never matter. The rule is simple -- if you see XML tag, you should have either field or the enum variant with that name. You should read [this chapter](https://docs.rs/quick-xml/latest/quick_xml/de/index.html#element-lists) to create correct mapping and understand why it should be that.

TL;DR:
```rust
#[derive(Serialize)]
struct A {
    name: String,
}

#[derive(Serialize)]
struct Tags {
    // #[serde(default)] - for deserialization of empty lists
    a: Vec<A>,
}

#[derive(Serialize)]
struct B {
    tags: Tags,
    name: String,
}
```
